### PR TITLE
Avatar component

### PR DESCRIPTION
**Pull request by patricklafrance**
Issue: https://github.com/gsoft-inc/sg-orbit/issues/535 https://github.com/gsoft-inc/sg-orbit/issues/668

## Summary

New Avatar and AvatarGroup components

## What I did

- Added a new Avatar component
- Added a new AvatarGroup component
- Added an AsyncImage component
- Added word-breaks and white space pre-wrap to components supporting rich content

## How to test

- Is this testable with Jest or Chromatic screenshots? Yes

- Does this need an update to the documentation? Yes

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

File: packages/react-components/src/avatar/docs/Avatar.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    githubPath="/packages/react-components/src/avatar/src"
+/>
+
+## Usage
+
+### Default
+
+By default, an avatar display the initials of a user.
+
+Since ShareGate's applications are very close to [Microsoft Office 365](https://www.office.com/), our avatar aims to be similar to O365 personas. Therefore, our initials avatar background colors match O365 personas.
```
Comment (fraincs) [inline packages/react-components/src/avatar/docs/Avatar.stories.mdx:36]: matches

Comment (fraincs) [inline packages/react-components/src/avatar/docs/Avatar.stories.mdx:36]: I would use are closely tied instead of very close

File: packages/react-components/src/avatar/docs/Avatar.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+            <Avatar size="lg" src={ChrisHadfield} name="Chris Hadfield" />
+            <Avatar size="xl" src={ChrisHadfield} name="Chris Hadfield" />
+            <Avatar size="2xl" src={ChrisHadfield} name="Chris Hadfield" />
+        </Inline>
+    </Story>
+</Preview>
+
+## AvatarGroup
+
+In some cases, you might want to stack avatars as a group.
```
Comment (fraincs) [inline packages/react-components/src/avatar/docs/Avatar.stories.mdx:84]: In some cases, when space is tight , you might want to stack avatars as a group.

File: packages/react-components/src/avatar/docs/Avatar.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+            <Avatar name="Sally Ride" />
+            <Avatar name="Alan Shepard" />
+            <Avatar src={ChrisHadfield} name="Chris Hadfield" />
+        </AvatarGroup>
+    </Story>
+</Preview>
+
+### Remainings
+
+An avatar group have a maximum of 4 slots. Once this threshold is exceeded, the remaining avatars are indicated as a count at the end.
```
Comment (fraincs) [inline packages/react-components/src/avatar/docs/Avatar.stories.mdx:98]: This can be interpreted as when there's more than 4 avatars we add a fifth one, I would rephrase it as : 

as the fourth slot.

File: packages/react-components/src/avatar/src/Avatar.css [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    --o-ui-avatar-sm: var(--o-ui-global-scale-golf);
+    --o-ui-avatar-md: var(--o-ui-global-scale-hotel);
+    --o-ui-avatar-lg: var(--o-ui-global-scale-india);
+    --o-ui-avatar-xl: var(--o-ui-global-scale-juliett);
+    --o-ui-avatar-2xl: var(--o-ui-global-scale-kilo);
+}
+
+.o-ui-avatar {
+    display: inline-flex;
+    border-radius: 100%;
```
Comment (fraincs) [inline packages/react-components/src/avatar/src/Avatar.css:13]: we have a variable for this `--o-ui-shape-circular`

File: packages/react-components/src/avatar/src/Avatar.css [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+/* GROUPS */
+.o-ui-avatar-group {
+    display: inline-flex;
+    align-items: center;
+    width: max-content;
+}
+
+/* GROUPS | REMAININGS */
+.o-ui-avatar-group-remainings {
+    background-color: var(--o-ui-global-cloud-50);
```
Comment (fraincs) [inline packages/react-components/src/avatar/src/Avatar.css:78]: How does this behaves in dark mode? A semantic colour would be ideal such as `--o-ui-alias-background-3`

File: packages/react-components/src/avatar/src/Avatar.css [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+}
+
+.o-ui-avatar-group-remainings .o-ui-avatar-text {
+    color: var(--o-ui-global-marine-400);
+}
+
+/* GROUPS | REMAININGS | SIZES */
```
Comment (fraincs) [inline packages/react-components/src/avatar/src/Avatar.css:85]: Should these be left here?

File: packages/react-components/src/avatar/src/Avatar.css [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+    width: max-content;
+}
+
+/* GROUPS | REMAININGS */
+.o-ui-avatar-group-remainings {
+    background-color: var(--o-ui-global-cloud-50);
+}
+
+.o-ui-avatar-group-remainings .o-ui-avatar-text {
+    color: var(--o-ui-global-marine-400);
```
Comment (fraincs) [inline packages/react-components/src/avatar/src/Avatar.css:82]: Same as comment above : How does this behaves in dark mode? If we use a semantic colour as a background we need to modify the text colour accordingly.

**Review by fraincs — APPROVED** [review]
Approved with changes

File: packages/react-components/src/avatar/src/Avatar.css (hunk 7 above)
Comment (patricklafrance) [reply packages/react-components/src/avatar/src/Avatar.css:82]: Would love too but I can't find any alias with this color?

Comment (patricklafrance) [reply packages/react-components/src/avatar/src/Avatar.css:82]: I went with `--o-ui-alias-text-2` which is equivalent to `--o-ui-global-marine-500`, hope it's fine.
